### PR TITLE
Grantnicholas/file based jdbc column mappings

### PR DESCRIPTION
**Pull request by grantatspothero**
Thanks @kokosing for the feedback on the original PR, was able to get the changes implemented without touching trino core/the SPI. 

I'm working through migrating the oracle tests to `BaseCaseInsensitiveMappingTest` as described here and running into some errors 
https://github.com/trinodb/trino/issues/7864

Would it be worth splitting the test migration bits into separate PRs instead of putting them in this PR? Having the tests migrated would be helpful to ensure the column mapping functionality works on different systems.

### CODE REVIEW

File: plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java [hunk 1]
```diff
@@ -204,11 +204,12 @@ protected boolean filterSchema(String schemaName)
                         Optional.empty(),
                         Optional.empty());
                 Optional<ColumnMapping> columnMapping = toColumnMapping(session, connection, typeHandle);
-                log.debug("Mapping data type of '%s' column '%s': %s mapped to %s", schemaTableName, columnName, typeHandle, columnMapping);
+                log.debug("Mapping data type of '%s' column '%s': %s mapped to %s", schemaTableName, remoteColumnName, typeHandle, columnMapping);
                 // skip unsupported column types
                 boolean nullable = (resultSet.getInt("NULLABLE") != columnNoNulls);
                 Optional<String> comment = Optional.ofNullable(emptyToNull(resultSet.getString("REMARKS")));
                 if (columnMapping.isPresent()) {
+                    String columnName = fromRemoteColumnName(session.getIdentity(), connection, getIdentifierMapping(), tableHandle, remoteColumnName);
```
Comment (grantatspothero) [inline plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:212]: This was actually a bug in the memsqlclient implementation `fromRemoteColumnName` was not being called.

Comment (ebyhr) [reply plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:212]: Could you send a new PR for bug fix? PostgreSQL connector might have the same issue.

**Review by hashhar — COMMENTED** [review]
I'm not sure this complexity is worth the benefits. It's very difficult to track which name is being used where. Another concern I have is that with this change quite a few of the critical path methods (query building and transformation) now involve lookups for the remote name.

Before this change it was reasonable to expect that ColumnHandle's represent remote columns while ColumnMetadata represents internal representation (with always lowercased names).

For table and schema mapping it's not as much of an issue since all the name conversions happen within the connector's implementation in very few methods.

A possible solution can be to instead refactor usages of ColumnHandle's and ColumnMetadata to ensure that any "engine" method uses ColumnMetadata (to always get the lowercased names) while "connector" methods use ColumnHandles (which must be changed to always store the remote name). And having a single method (in BaseJdbcClient) that is responsible for converting the ColumnMetadata back to a ColumnHandle with remote names. That method can then be used where "connector" needs to use information provided to it via "engine".

**Comment by grantatspothero** [comment]
> A possible solution can be to instead refactor usages of ColumnHandle's and ColumnMetadata to ensure that any "engine" method uses ColumnMetadata (to always get the lowercased names) while "connector" methods use ColumnHandles (which must be changed to always store the remote name). And having a single method (in BaseJdbcClient) that is responsible for converting the ColumnMetadata back to a ColumnHandle with remote names. That method can then be used where "connector" needs to use information provided to it via "engine".

I agree with the idea of JdbcColumnHandle storing remote names and ColumnMetadata storing trino names, but there are some problems with this approach:
1. Converting from/to remote column names is not invertible, meaning it is not sufficient to simply have ColumnMetadata store a trino name and then call `toRemoteColumnName` on the trino name when the remote name is needed:

example using DefaultIdentifierMapping:
```
remoteColumn="MixedCase"
fromRemoteColumnName("MixedCase") == "mixedcase"
toRemoteColumnName(fromRemoteColumnName("MixedCase") ) == "MIXEDCASE"
```
This breaks connectors whose underlying database is case sensitive (for example an oracle table defined with delimited columns). 


2. Due to the above, this means we need to refactor all methods that interact with the remote db to take a ColumnHandle param instead of a ColumnMetadata param:

however, the ConnectorMetadata interface defines `addColumn` as taking a `ColumnMetadata` not a `ColumnHandle` so this refactor would break an interface in the spi.
https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java#L402-L405

3. Since (2) breaks the spi, we could try to convert a trino name into a remote name by querying the external system again for metadata `getColumns(ConnectorSession session, JdbcTableHandle tableHandle)` inside each method that needs the remote name but only has a `ColumnMetadata`. However this adds an extra unnecessary metadata query against the remote db for those operations (like add column):

For example here: https://github.com/trinodb/trino/blob/master/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java#L711-L720

4. Since (3) adds extra metadata queries, we could try to make the from/to remote column name methods invertible. However, the only way I can think of doing this is storing a map from `(schema, table, trinoColumnName) -> remoteColumnName` for every single column which will consume a large amount of memory and seemss unfeasible (how do entries get evicted?)

**Comment by wendigo** [comment]
Let's hold on working on this task. It seems harder than I anticipated, and introducing that complexity outweighs desired benefits.

**Pull request closed without merging**